src/rust/ide/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -549,18 +550,24 @@ impl Container {
             eval  frp.set_size          ((s) size.set_target_value(*s));
 
             mouse_down_target <- scene.mouse.frp.down.map(f_!(scene.mouse.target.get()));
-            eval mouse_down_target ([model] (target){
+            selected <- mouse_down_target.map(f!([model] (target){
                 let vis        = &model.visualization;
                 let activate   = || vis.borrow().as_ref().map(|v| v.activate.clone_ref());
                 let deactivate = || vis.borrow().as_ref().map(|v| v.deactivate.clone_ref());
                 if model.is_this_target(*target) {
                     if let Some(activate) = activate() {
                         activate.emit(());
+                        return Some(true);
                     }
                 } else if let Some(deactivate) = deactivate() {
                     deactivate.emit(());
+                    return Some(false);
                 }
-            });
+                None
+            })).unwrap();
+
+            is_selected_changed    <- selected.map2(&frp.source.is_selected, |&new,&old| if new != old {Some(new)} else {None}).unwrap();
```

Comment:
1. you should use `frp.source` only for setting values, never for reading them back
2. Line too long
3. Use `<=` instead of unwrap
4. Use `(new != old).as_some(new)`